vuu-ui/packages/vuu-shell/src/application-settings/ApplicationSettingsPanel.tsx

```diff
@@ -1,211 +1,28 @@
-  defaultValue?: string;
-  type: "string";
-}
-export interface NumericProperty extends BaseProperty {
-  values?: number[] | Option<number>[];
-  defaultValue?: number;
-  type: "number";
-}
-export interface BooleanProperty extends BaseProperty {
-  defaultValue?: boolean;
-  type: "boolean";
-}
-
-export type SettingsProperty =
-  | StringProperty
-  | NumericProperty
-  | BooleanProperty;
-
-export const isBooleanProperty = (
-  property: SettingsProperty
-): property is BooleanProperty => property.type === "boolean";
-
-export const isStringOrNumber = (value: unknown): value is string | number =>
-  typeof value === "string" || typeof value === "number";
-
-export interface SettingsSchema {
-  properties: SettingsProperty[];
-}
-
-const getValueAndLabel = (value: string | number | Option<string | number>) =>
-  isOption(value) ? [value.value, value.label] : [value, value];
-
-// Determine the form control type to be displayed
-export function getFormControl(
-  property: SettingsProperty,
-  changeHandler: FormEventHandler,
-  selectHandler: DropdownProps["onSelectionChange"],
-  currentValue?: string | boolean | number
-) {
-  if (isBooleanProperty(property)) {
-    const checked =
-      typeof currentValue === "boolean"
-        ? currentValue
-        : property.defaultValue ?? false;
-
-    return (
-      <Switch
-        checked={checked}
-        label={property.label}
-        onChange={changeHandler}
-      ></Switch>
-    );
-  }
-  // Toggle Box for 1 or 2 values
-  if (Array.isArray(property.values)) {
-    if (property.values.length <= 2) {
-      return (
-        <ToggleButtonGroup
-          value={currentValue as ToggleButtonGroupProps["value"]}
-          onChange={changeHandler}
-        >
-          {property.values.map((valueOrOption) => {
-            const [value, label] = getValueAndLabel(valueOrOption);
-            return (
-              <ToggleButton key={value} value={value}>
-                {label}
-              </ToggleButton>
-            );
-          })}
-        </ToggleButtonGroup>
-      );
-    } else if (property.values.length > 2) {
-      return (
-        <Dropdown
-          value={currentValue as DropdownProps["value"]}
-          onSelectionChange={selectHandler}
-        >
-          {property.values.map((valueOrOption) => {
-            const [value, label] = getValueAndLabel(valueOrOption);
-            return (
-              <Option value={value} key={value} data-field={property.name}>
-                {label}
-              </Option>
-            );
-          })}
-        </Dropdown>
-      );
-    }
-  } else {
-    const value = isStringOrNumber(currentValue)
-      ? currentValue
-      : isStringOrNumber(property.defaultValue)
-      ? property.defaultValue
-      : "";
-    return <Input value={value} />;
-  }
-}
+import { useApplicationSettings } from "../application-provider";
 
 // Props for Panel
 export interface ApplicatonSettingsPanelProps
```

Comment:
because ApplicationSettingsPanel no longer implements any props of its own (just inherits from HTMLDivEl;ement), no need to specify a type for the props, just use `HTMLAttributes<HTMLDivElement>` directly as the type.

If you DID want to declare a type, you would use `type` rather than `interface`, so

```
export  type ApplicatonSettingsPanelProps = HTMLAttributes<HTMLDivElement>;
```
this is just declaring an alias that doesn't add any new props.